### PR TITLE
CI: Lint docs

### PR DESCRIPTION
**Pull request by szepeviktor**
There is a tool called `sphinx-lint`

### CODE REVIEW

File: doc/rules/whitespace/indentation_type.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Example #1
     <?php
 
     if (true) {
-   -	echo 'Hello!';
+   -<TAB>echo 'Hello!';
```
Comment (szepeviktor) [inline doc/rules/whitespace/indentation_type.rst:20]: reStructuredText converts TAB characters to spaces :(

**Comment by szepeviktor** [comment]
❔ Whet to do with two dozen `line-too-long`-s?

**Comment by Wirone** [comment]
Yeah, this one may be tricky, since docs are generated (with `composer docs` script), and there are tests that verify consistency between source code and generated RSTs. But it looks promising, it's great to spot such things like in this PR's diff 👌.

**Comment by szepeviktor** [comment]
@Wirone Please help me code these two.
```
doc/ruleSets/PER-CS1.0.rst:5: default role used (hint: for inline literals, use double backticks) (default-role)
doc/ruleSets/PER-CS1.0Risky.rst:5: default role used (hint: for inline literals, use double backticks) (default-role)
```

**Comment by Wirone** [comment]
@szepeviktor I am AFK right now, reviewing from mobile. Now I am going to be mostly offline, I can look in the evening if the problem remains 🙂.

**Comment by Wirone** [comment]
@szepeviktor I've changed those 2 sets locally (double backtick), generated docs (`composer docs`) and whole QA suite (`composer qa`) is green. I've reproduced Sphinxlint's CI job locally (inside of `docker run --rm -it -w /fixer -v `pwd`:/fixer ubuntu:22.04 bash`) and it looks OK:

```shell
root@996bbd4a9326:/fixer# apt-get update && apt-get install python3 pip git && pip install --user sphinx-lint && mv /root/.local/bin/sphinx-lint /usr/local/bin/

root@996bbd4a9326:/fixer# git --version
git version 2.34.1
root@996bbd4a9326:/fixer# sphinx-lint --version
sphinx-lint 0.9.0
root@996bbd4a9326:/fixer# git ls-files --cached -z -- '*.rst' | xargs --null -- python3 -m sphinxlint --enable all --disable trailing-whitespace --max-line-length 2000
No problems found.
```

I'll just push the changes here (with a rebase).

File: .github/workflows/sphinx.yml
Comment (Wirone) [inline .github/workflows/sphinx.yml:1]: @keradus maybe we should group all linters in one workflow? I mean, instead of having `spinhx.yml` and `yaml.yml`, maybe we should have `lint.yml` and put all linters there, so instead:

image

we would have:

image

?

It's not a blocker here, probably it should be done in separate PR (that I reserve for myself 😉). I just want to discuss if this is good direction.

**Review by Wirone — APPROVED** [review]
Great stuff @szepeviktor 👏 . LGTM.

File: .github/workflows/sphinx.yml
Comment (szepeviktor) [reply .github/workflows/sphinx.yml:1]: My pyramid goes like
- workflow: **What** are we checking?
- job: The check
- step: Atomic operations


https://github.com/szepeviktor/byte-level-care/tree/master/.github/workflows

Comment (Wirone) [reply .github/workflows/sphinx.yml:1]: I find `YAML Validation / Validate YAML` redundant. What I see here as a common pattern is linting (the "what" part), then jobs can make it more precise:

- `Lint / YAML`
- `Lint / reStructuredText`
- `Lint / Markdown` (not existing now, but can be added)
- `Lint / Dockerfile` (not existing now, but can be added)
- `Lint / Pull Request name`
etc.

I am more of a Gitlab user and this is something I would do there, several `lint:*` jobs.

Comment (szepeviktor) [reply .github/workflows/sphinx.yml:1]: The "verb "lint" is on viktor's blacklist as is a slang.

Comment (szepeviktor) [reply .github/workflows/sphinx.yml:1]: These non source code files are equally important for me as I care about running things not building things.

Comment (szepeviktor) [reply .github/workflows/sphinx.yml:1]: But I let you go on your _linting_ way!!

Comment (Wirone) [reply .github/workflows/sphinx.yml:1]: I understand your point of view @szepeviktor, I'm just thinking out loud and want to find a way that fits @keradus' vision and at the same time looks good from my perspective 😉. In terms of "linting", in my opinion it's so established term that can't be described as "slang", and even if it's not official enough, it just describes the intent properly, without any room for misinterpretation.

Comment (szepeviktor) [reply .github/workflows/sphinx.yml:1]: I think Greg's linting = syntax check + formatting

Comment (Wirone) [reply .github/workflows/sphinx.yml:1]: Not only that. Fixer itself can act as linter (with `php-cs-fixer check`, which is an alias for `fix --dry-run`), it can find not only formatting issues, but also other violations (the scope depends on the ruleset). I believe linting is generic (but still descriptive enough) name for operation that verifies code against some rules. These rules can be focused on formatting, but also on good practices - it does not matter that much, when it comes to understanding what "linting" is. But that's only my opinion 🙂.

Comment (Wirone) [reply .github/workflows/sphinx.yml:1]: Discussion moved to #7485.
